src/tools/tidy/src/edition.rs

```diff
@@ -11,7 +11,10 @@ fn is_edition_2021(mut line: &str) -> bool {
 pub fn check(path: &Path, bad: &mut bool) {
     walk(
         path,
-        &mut |path| filter_dirs(path) || path.ends_with("tests"),
+        &mut |path| {
+            filter_dirs(path)
+                || (path.ends_with("tests") && path.join("COMPILER_TESTS.md").exists())
```

Comment:
this `COMPILER_TESTS.md` change is new, right? why did that change?

If you're trying to filter *only* the top-level directory, maybe you can do that directly?
```suggestion
                || path == "tests"
```

Reply:
This doesn't work, because the path will never start with `tests`. It will at least start with a `/` or `./`. I do not know how to detect this any better

Reply:
While investigating this issue, I just saw that I forgot to also include the tests directory for this check, thanks!

Here is an example of path this function gets: `/home/gh-albertlarsan68/rust/compiler/rustc_arena/src/tests.rs`, and they are all like this, so testing for exact match with `tests` does definitely not work.